### ISSUE TITLE
Opening Brace false positive when closure is returned without return statement (Swift 5.1)

### ISSUE THREAD

**Issue by antiraum** [realm/SwiftLint#2914]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

When returning a closure without a return statement, as it's possible since Swift 5.1, the opening brace rule reports a false positive.

```swift
/// hello
public var closure: () -> Void {
    {
        print("hello")
    }
}
```

is autocorrected to 

```swift
/// hello
public var closure: () -> Void { {
        print("hello")
    }
}
```

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint --no-cache test.swift 
Linting Swift files at paths test.swift
Linting 'test.swift' (1/1)
/Users/tom/test.swift:3:5: warning: Opening Brace Spacing Violation: Opening braces should be preceded by a single space and on the same line as the declaration. (opening_brace)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?

0.35.0

* Installation method used (Homebrew, CocoaPods, building from source, etc)?

Homebrew

* Which Xcode version are you using (check `xcode-select -p`)?

11.1

* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

see above

**Comment by BasThomas** [comment]
Also linking adding support for implicit returns here: https://github.com/realm/SwiftLint/issues/2870

**Comment by iliaskarim** [comment]
The corrected code looks fine.

**Comment by Fryie** [comment]
could we make this at least configurable?
